File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums, int l, int r){

        if(l>=r){
            return nums[r];
        }

        int min;
        int mid = (l+r)/2;

        return min = Math.min(findMin(nums, l, mid),findMin(nums, mid+1, r));
    }
    public int findMin(int[] nums) {

        int l = 0;
        int min;
        int r = nums.length-1;

        int mid = (l+r)/2;

        return min = Math.min(findMin(nums, l, mid),findMin(nums, mid+1, r));
        
    }
}

File: Data Structures & Algorithms/minimum-difference-between-highest-and-lowest-of-k-scores/submission-0.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        print(nums)
        l = 0
        r = k-1
        n = len(nums)

        diff = nums[r] - nums[l]

        while(r <n-1):
            l += 1
            r += 1
            diff = min(diff, nums[r] - nums[l])

        return diff

        

